### PR TITLE
CASSANDRA-17122 trunk: Add utilities for materialized views to CQLTester

### PR DESCRIPTION
**Pull request by adelapena**

### CODE REVIEW

File: test/long/org/apache/cassandra/cql3/ViewLongTest.java [hunk 1]
```diff
@@ -203,13 +142,14 @@ public void runMayThrow()
 
         if (rows.size() > 1)
         {
-            throw new AssertionError(String.format("Expected 1 row, but found %d; %s c = %d, and (%s) of which (%s) failed to insert", rows.size(), containsC ? "found row with" : "no rows contained", value, others, overlappingFailedWrites));
+            throw new AssertionError(String.format("Expected 1 row, but found %d; %s c = %d, and (%s) of which (%s) failed to insert",
```
Comment (bereng) [inline test/long/org/apache/cassandra/cql3/ViewLongTest.java]: formatting

File: test/unit/org/apache/cassandra/cql3/CQLTester.java [hunk 2]
```diff
@@ -861,6 +876,77 @@ protected void createTableMayThrow(String query) throws Throwable
...
+    {
+        return createView(null, query);
+    }
+
+    protected String createView(String viewName, String query)
+    {
+        String currentView = viewName == null ? createViewName() : viewName;
+        String fullQuery = String.format(query, KEYSPACE + "." + currentView, KEYSPACE + "." + currentTable());
+        logger.info(fullQuery);
+        schemaChange(fullQuery);
```
Comment (bereng) [inline test/unit/org/apache/cassandra/cql3/CQLTester.java:928]: I wonder if we should call here `waitForViewBuild()` bc I can see this being a future source of flakiness...

**Comment by bereng** [comment]
Well I can't follow down to the last detail de 1500 lines of change lol But overall lgtm. I left a nit and one I think it is important. Wdyt?

File: test/long/org/apache/cassandra/cql3/ViewLongTest.java (hunk 1 above)
Comment (adelapena) [reply test/long/org/apache/cassandra/cql3/ViewLongTest.java]: +1, done

File: test/unit/org/apache/cassandra/cql3/CQLTester.java (hunk 2 above)
Comment (adelapena) [reply test/unit/org/apache/cassandra/cql3/CQLTester.java:928]: Good idea, that could avoid future mistakes and makes the current tests a bit more concise, done [here](https://github.com/apache/cassandra/pull/1306/commits/464ca8473f5301a14e2f7f57a038e490c84ac7b6).

There are a couple of tests in `ViewTest` (`testViewBuilderResume` and `testTruncateWhileBuilding`) that rely on `createView` not waiting for the build tasks. For those special cases I have added a `CQLTester#createViewAsync` method that keeps the previous behaviour.

**Comment by adelapena** [comment]
@bereng Thanks for the review, I have tried to incorporate your suggestions. CI for [j8](https://app.circleci.com/pipelines/github/adelapena/cassandra/1131/workflows/fcd2d95e-c841-4db9-93e9-6832a48eb1bb) and [j11](https://app.circleci.com/pipelines/github/adelapena/cassandra/1131/workflows/fec57435-a8c0-42d3-80fb-455e4d2c882f) contains more repeated runs.

**Comment by bereng** [comment]
+1

**Pull request closed without merging**